src/ast/semantic_analyser.cpp

```diff
@@ -277,34 +277,38 @@ void SemanticAnalyser::visit(Call &call)
     call.type = SizedType(Type::count, 8);
   }
   else if (call.func == "sum") {
+    bool sign = false;
     check_assignment(call, true, false);
-    check_nargs(call, 1);
-
-    call.type = SizedType(Type::sum, 8);
+    if (check_nargs(call, 1)) {
+      sign = call.vargs->at(0)->type.is_signed;
+    }
```

Comment:
The if is required because `at(0)->type` is invalid if no args are passed. 

The idea here is to copy the sign of the argument (it only taks one). Although I do not have a use case I could imagine some one summing large uints and having that print negative numbers would be annoying  so I would be nice if that can be avoided. But looking at it again I don't think the current implementation is good enough as it doesn't prevent you from mixing both and  getting unexpected results. So maybe it is better to make them signed and document them as such? If someone has a use case for unsigned it can be added later.

Reply:
I think it works. In which case would it be a problem?

Btw, the question was not limited for this if block on `sum`, as the same if block is present on other the other functions as well. Thanks for the answer, makes sense :)

Reply:
Indeed,  min/max/sum/avg/stats where all done with the same reasoning.

What I mean is that this works:

```
$ sudo bpftrace -e 't:syscalls:sys_exit_openat / comm == "cat" && args->ret < 0 / { printf("hit\n"); @=sum(args->ret); }'

@: -14
```

And this works:

```
$ sudo bpftrace -e 'kr:sock_recvmsg /comm == "curl"/ { @=sum(retval); printf("hit\n"); }'

@: 978
```

But combining the two programs will give you either `@: 18446744073709548282` or `@: -12308` depending on whether you did A+B or B+A.

We should probably have `SemanticAnalyser::assign_map_type` emit a warning or error for this when a user does this. But that means adding more code to this MR which I rather not do. I think we could either leave it as is and add checking for this later, or go to signed for now and implement unsigned later.